pandas/core/indexes/extension.py

```diff
@@ -170,6 +172,29 @@ class ExtensionIndex(Index):
     __le__ = _make_wrapped_comparison_op("__le__")
     __ge__ = _make_wrapped_comparison_op("__ge__")
 
+    def __getitem__(self, key):
+        result = self._data[key]
+        if isinstance(result, type(self._data)):
+            return type(self)(result, name=self.name)
```

Comment:
Should we use a faster constructor (simple_new ?) when we just want to wrap the correct type of ExtensionArray in the index?

Reply:
I think that'd work.  IIRC there were some corner cases involving CategoricalIndex.dtype, not sure if those are relevant here